src/format.ts

```diff
@@ -25,8 +25,13 @@ export function format(
   prettier_options: Options = {},
   { reporter = Reporter.Warning, filter = false }: PluginOptions = {},
 ) {
-  return create_transform((text, filename) => {
+  return create_transform(async (text, filename) => {
+    const resolved_config = await (typeof prettier.resolveConfig === 'function'
+      ? prettier.resolveConfig(filename)
+      : Promise.resolve(null));
+
     const formatted = prettier.format(text, {
+      ...resolved_config,
```

Comment:
I'd like to either use `resolved_config` or `prettier_config` rather than mix them together, what do you think?

Reply:
I was taking my cues from the addition of `--config-precedence` in [Prettier 1.7.0](https://github.com/prettier/prettier/releases/tag/1.7.0).  To me, it seems like the least surprising implementation, but I'm open to changing if you disagree.

Reply:
Agree, but it'd be better to also have an option to disable the `resolved_config`, so that people can choose whatever they want:

- use `resolved_config` + `prettier_options` by default
- `pluginOptions.configFile: false` to ignore `resolved_config`